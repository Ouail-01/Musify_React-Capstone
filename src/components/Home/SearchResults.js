/* eslint-disable */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { changeMusic } from '../../redux/music/musicSlice';
import { fetchAlbums } from '../../redux/music/albumSlice';

export default function SearchResults({ id }) {
  SearchResults.propTypes = {
    id: PropTypes.string.isRequired,
  };

  const dispatch = useDispatch();

  const data = useSelector((state) => state.albums.data);
  const status = useSelector((state) => state.albums.status);

  const handleSongClick = (item) => {
    dispatch(changeMusic(item));
  };

  useEffect(() => {
    dispatch(fetchAlbums(id));
  }, [id]);

  if (status === 'loading') {
    return (
      <div className="loading">
        <div className="lds-roller">
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    );
  }

  if (status === 'failed') {
    return <div>Oops! Something went wrong.</div>;
  }

  return (
    <>
      <div className="results">
        {data.data
          && data.data.map((item) => (
            <div
              key={item.id}
              className="song"
              onClick={() => handleSongClick(item)}
              onKeyDown={(event) => {
                if (event.key === 'Enter') {
                  handleSongClick(item);
                }
              }}
              role="button"
              tabIndex={0}
            >
              <img src={item.album.cover_big} alt={item.title} />
              <div>
                <h1>{item.title}</h1>
                <p>{item.artist.name}</p>
              </div>
            </div>
          ))}
      </div>
    </>
  );
}
