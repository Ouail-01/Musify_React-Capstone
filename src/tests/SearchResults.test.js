/* eslint-disable */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import SearchResults from '../components/Home/SearchResults';
import store from '../redux/configureStore';

describe('SearchResults component', () => {
  it('renders correctly', () => {
    const props = {
      id: '123',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <SearchResults {...props} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
