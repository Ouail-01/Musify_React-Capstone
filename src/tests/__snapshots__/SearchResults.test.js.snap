// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="lds-roller"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
